scripts/diff_against_dev.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash 
+
+## This file will do a diff against the dev branch for added files
+## As the definition of the versioning is to always create a new version if a file has been changed
+## we make sure that every structure change is a breaking change. 
+##
+## But at the same time it's hard to see the actual changes in the new version as there is no 
+## history of changes. Therefore this script exists which will:
+##
+## * Get the *added* files out of the git history
+## * Extract the version 
+## * Check if a file with version-1 exists in the origin branch
+## * Do a diff against the other file
+
+CUR_BRANCH=$(git branch --show-current)
+ORIGIN=${1:-dev}
+
+git fetch origin $ORIGIN
+
+function check_file {
+	FILE=$1
```

Comment:
`$1` is used for both the `ORIGIN` and `FILE` variables?

Reply:
$1 in the function refers to the parameter passed to the function not the script parameter

Reply:
🤦🏼‍♂️ my dislike to bash scripting shows